### PR TITLE
Atomic tuning table

### PR DESCRIPTION
**Pull request by marcussatellite**
Do not merge...need review by @sieren .

This PR removes all swift dependencies from S1DSPKernel, except for S1Sequencer.

This could resolve some crash reports.

### CODE REVIEW

File: AudioKitSynthOne/DSP/Audio Unit/S1AudioUnit.h [hunk 1]
```diff
@@ -101,6 +97,16 @@ typedef struct S1ArpBeatCounter {
 - (void)resetDSP;
 - (void)resetSequencer;
 
+// S1TuningTable protocol
+- (void)setTuningTable:(float)frequency index:(int)index;
+- (float)getTuningTable:(int)index;
```
Comment (sieren) [inline AudioKitSynthOne/DSP/Audio Unit/S1AudioUnit.h]: I'd rephrase this to `getTuningTableFrequency:(int)noteIndex;` .. others might be confused else.

File: AudioKitSynthOne/DSP/Kernel/S1DSPKernel.mm [hunk 2]
```diff
@@ -95,6 +97,13 @@
     }
     _rate.init();
 
+    // intialize dsp tuning table with 12ET
+    for(int i = 0; i < 128; i++) {
+        tuningTable[i].store(440. * exp2((i - 69)/12.));
+    }
+    tuningTableNPO.store(12);
```
Comment (sieren) [inline AudioKitSynthOne/DSP/Kernel/S1DSPKernel.mm]: What's this magic number?

File: AudioKitSynthOne/DSP/Kernel/S1DSPKernel.mm [hunk 3]
```diff
@@ -139,7 +148,28 @@
...
+// private tuningTable lookup
+double S1DSPKernel::tuningTableNoteToHz(int noteNumber) {
+
+    const int nn = clamp(noteNumber, 0, 127);
+    return getTuningTable(nn);
+}
+
+// S1TuningTable protocol
+void S1DSPKernel::setTuningTable(float frequency, int index) {
+    tuningTable[index].store(frequency);
```
Comment (sieren) [inline AudioKitSynthOne/DSP/Kernel/S1DSPKernel.mm]: Is it worth clamping the index here as well like above?

File: AudioKitSynthOne/DSP/Kernel/S1DSPKernel.hpp [hunk 4]
```diff
@@ -220,7 +220,18 @@ class S1DSPKernel : public AKSoundpipeKernel, public AKOutputBuffered {
     
     float monoFrequencySmooth = 261.6255653006f;
 
+    // S1TuningTable protocol
+    void setTuningTable(float value, int index);
+    float getTuningTable(int index);
+    void setTuningTableNPO(int npo);
+
 private:
+    std::array<std::atomic<float>, 128> tuningTable;
```
Comment (sieren) [inline AudioKitSynthOne/DSP/Kernel/S1DSPKernel.hpp]: I wonder if we should default initialize the `std array` somewhere just for safety, maybe in the init function by calling `tuningTable.fill(0);` or whatever number makes sense?

File: AudioKitSynthOne/DSP/Audio Unit/S1AudioUnit.h (hunk 1 above)
Comment (marcussatellite) [reply AudioKitSynthOne/DSP/Audio Unit/S1AudioUnit.h]: Will update

File: AudioKitSynthOne/DSP/Kernel/S1DSPKernel.mm (hunk 2 above)
Comment (marcussatellite) [reply AudioKitSynthOne/DSP/Kernel/S1DSPKernel.mm]: Legacy default magic numbers for backwards compatibility.  Will be overwritten when the application layer tuning model initializes, which is pretty fast.

File: AudioKitSynthOne/DSP/Kernel/S1DSPKernel.mm (hunk 3 above)
Comment (marcussatellite) [reply AudioKitSynthOne/DSP/Kernel/S1DSPKernel.mm]: Yes

File: AudioKitSynthOne/DSP/Kernel/S1DSPKernel.hpp (hunk 4 above)
Comment (marcussatellite) [reply AudioKitSynthOne/DSP/Kernel/S1DSPKernel.hpp]: 1 is a reasonable default

Comment (marcussatellite) [reply AudioKitSynthOne/DSP/Kernel/S1DSPKernel.hpp]: 1 is a reasonable default

File: AudioKitSynthOne/DSP/Kernel/S1DSPKernel.mm (hunk 2 above)
Comment (sieren) [reply AudioKitSynthOne/DSP/Kernel/S1DSPKernel.mm]: You could do that where you declare the member variable too.. `std::atomic<int> tuningTableNPO{12};` saves a line of code :)

**Comment by sieren** [comment]
I've added it to the Sequencer btw: https://github.com/marcussatellite/AudioKitSynthOne/pull/1 feel free to merge into your branch :)

**Comment by aure** [comment]
👍

File: AudioKitSynthOne/DSP/Kernel/S1DSPKernel.hpp (hunk 4 above)
Comment (marcussatellite) [reply AudioKitSynthOne/DSP/Kernel/S1DSPKernel.hpp]: done

**Comment by marcussatellite** [comment]
I addressed the feedback on this PR
